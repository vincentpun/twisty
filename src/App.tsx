import * as React from 'react';
import { hot } from 'react-hot-loader';
import { Provider } from 'react-redux';
import { Store } from 'redux';

import AppControl from './components/AppControl';

const App = ({ store }: { store?: Store }) => (
  <Provider store={store}>
    <div className="row no-gutters">
      <AppControl productName="Twisty" />
      <div className="col">
      </div>
    </div>
  </Provider>
);

export default hot(module)(App);
