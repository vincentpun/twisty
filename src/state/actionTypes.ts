import { UIActionTypes } from './ui/actionTypes';

export type ActionTypes = UIActionTypes;
