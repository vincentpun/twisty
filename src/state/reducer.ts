import { combineReducers } from 'redux';

import ui, { UIState } from './ui/reducer';

export interface State {
  ui: Readonly<UIState>
}

export default combineReducers<State>({
  ui,
});
