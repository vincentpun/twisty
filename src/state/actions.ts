import { ActionTypes } from './actionTypes';

export interface Action<T extends ActionTypes> {
  type: T;
}

export interface ActionWithPayload<T extends ActionTypes, P> {
  type: T;
  payload: P;
}

export function createAction<T extends ActionTypes>(type: T): Action<T>
export function createAction<T extends ActionTypes, P>(type: T, payload: P): ActionWithPayload<T, P>
export function createAction<T extends ActionTypes, P>(type: T, payload?: P) {
  return (payload === undefined) ? { type } : { type, payload };
}

type Function = (...args: any[]) => any;
type ActionCreatorsMap = { [key: string]: Function };

export type ActionUnion<A extends ActionCreatorsMap> = ReturnType<A[keyof A]>;
