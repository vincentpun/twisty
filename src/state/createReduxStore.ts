import { applyMiddleware, compose, createStore, StoreEnhancer } from 'redux';
import thunkMiddleware from 'redux-thunk';

import reducer from './reducer';

export const createReduxStore = (initialState?: any) => {
  const isBrowser = typeof window === 'object';
  const devToolsExtension: StoreEnhancer = isBrowser &&
    ((window as any)['devToolsExtension'] ? (window as any)['devToolsExtension']() : f => f);

  const middlewares = [thunkMiddleware];

  const enhancers = [
    applyMiddleware(...middlewares),
    devToolsExtension,
  ].filter(Boolean);

  return createStore(
    reducer,
    initialState,
    compose(...enhancers),
  );
};
