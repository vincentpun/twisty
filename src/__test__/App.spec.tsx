import * as React from 'react';
import { shallow } from 'enzyme';

import App from '../App';
import { createReduxStore } from '../state/createReduxStore';

const store = createReduxStore();

describe('App', () => {
  it('matches the snapshot', () => {
    const wrapper = shallow(<App store={store} />);
    expect(wrapper).toMatchSnapshot();
  });
});
