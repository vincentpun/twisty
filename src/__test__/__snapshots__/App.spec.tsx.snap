// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <div
    className="row no-gutters"
  >
    <AppControl
      productName="Twisty"
    />
    <div
      className="col"
    />
  </div>
</Provider>
`;
